cmd/execute.go

```diff
@@ -186,13 +190,29 @@ func watchEthStorage(w watcher.IStorageWatcher, wg *syn.WaitGroup) {
 	defer wg.Done()
 	// Execute over the StorageTransformerInitializer set using the storage watcher
 	LogWithCommand.Info("executing storage transformers")
-	ticker := time.NewTicker(pollingInterval)
-	defer ticker.Stop()
-	for range ticker.C {
-		errs := make(chan error)
-		diffs := make(chan storageUtils.StorageDiff)
-		w.Execute(diffs, errs, queueRecheckInterval)
+	on := viper.GetBool("storageBackFill.on")
+	if on {
+		go backFillStorage(w)
+	}
+	w.Execute(queueRecheckInterval, on)
```

Comment:
In our IPM today, we talked about how the backfill will block the regular syncing to make sure that the diffs come in in the correct order - is this the place that does that? Does `go backFillStorage(w)` block it's parent go routine?

Reply:
I'm not sure the iteration where the backfill retained order of the diffs was ever up here tbh, the way that worked though was to block before this [processRow call](https://github.com/vulcanize/vulcanizedb/blob/statediff_at/libraries/shared/watcher/storage_watcher.go#L97) by listening for the "done" signal from the backfill process, which has now been moved to a [non-blocking case](https://github.com/vulcanize/vulcanizedb/blob/statediff_at/libraries/shared/watcher/storage_watcher.go#L100).

Reply:
Isn't backfill necessarily delivering things in the wrong order? Or is the idea to *only* backfill before even starting to sync any newer contract data? That needs to be made very clear if so.

Reply:
Yeah the backfill and forward sync happen concurrently much like how the header sync progresses at the head and tail. I can confirm the geth node publishes to both channels concurrently from my tests with the super node backfill which uses the same patterns, but obviously this still needs to be tested in the same capacity once arch1 is setup. The backfill itself does not guarantee any ordering.